### PR TITLE
Treat sphinx warnings as errors in travis

### PR DESCRIPTION
**Pull request by joshmoore**
Fixes applied:

 * Corrected formatting of anchors
 * Added a few missing pages to index
 * Un-ref'd a few undocumented classes
 * Allow overriding Makefile variables

Test plan:
 * If travis is green, then in the future doc warnings should be avoided.
 * Docs should also look reasonable.

### CODE REVIEW

File: docs/source/debugging.rst [hunk 1]
```diff
@@ -22,7 +22,7 @@ data that was updated to work post-update:
     + http://spacetx.starfish.data.public.s3.amazonaws.com/browse/formatted/20180926/iss_breast/experiment.json
 
 If you're using your own data with starfish, you may need to re-run your data ingestion workflow
-based on :ref:`TileFetcher` and :ref:`FetchedTile` to generate up-to-date versions of spaceTx-format.
+based on ``TileFetcher`` and ``FetchedTile`` to generate up-to-date versions of spaceTx-format.
```
Comment (ttung) [inline docs/source/debugging.rst]: is this supposed to be ``:py:class:`starfish.experiment.builder.providers.TileFetcher` `` and ``:py:class:`starfish.experiment.builder.providers.FetchedTile` ``

File: docs/source/usage/iss/iss_cli.sh [hunk 2]
```diff
@@ -54,7 +54,7 @@ starfish target_assignment \
     --coordinates-geojson /tmp/starfish/results/regions.geojson \
     --intensities /tmp/starfish/results/spots.nc \
     --output /tmp/starfish/results/targeted-spots.nc \
-    PointInPoly2D
+    Label
```
Comment (ttung) [inline docs/source/usage/iss/iss_cli.sh]: @shanaxel42 isn't this tested as part of unit tests?

Comment (joshmoore) [reply docs/source/usage/iss/iss_cli.sh]: There's a python version that was updated in https://github.com/spacetx/starfish/commit/b1c20bcef23c23df46ae809c7d31950d8b1f4f1b#diff-82b3014b798b61630569e45cc9b33581R117

**Comment by joshmoore** [comment]
Thanks for the :py:class suggestion. Landing. Note: open PRs may have doc changes that will lead to a failure on master when they are merged.

**Pull request merged**